### PR TITLE
yoav - Rotate Around Wheel (6 files) (Don't merge)

### PR DESCRIPTION
**Pull request by YHerm**

### CODE REVIEW

File: src/main/java/frc/robot/subsystems/swerve/ModuleUtils.java [hunk 1]
```diff
@@ -20,6 +21,15 @@ public static double toDriveDistance(double revolutions) {
         return Conversions.revolutionsToDistance(revolutions, ModuleConstants.WHEEL_DIAMETER_METERS);
     }
 
+    public static Translation2d getModulePositionRelativeToMiddleOfRobot(ModuleName moduleName){
+        return switch (moduleName) {
+            case FRONT_LEFT ->  new Translation2d(SwerveConstants.FRONT_LEFT_TRANSLATION2D.getX(), SwerveConstants.FRONT_LEFT_TRANSLATION2D.getY());
```
Comment (YoniKiriaty) [inline src/main/java/frc/robot/subsystems/swerve/ModuleUtils.java:26]:
```suggestion
            case FRONT_LEFT ->  new Translation2d(SwerveConstants.FRONT_LEFT_TRANSLATION2D);
```

im pretty sure you can do this

File: src/main/java/frc/robot/subsystems/swerve/Swerve.java [hunk 2]
```diff
@@ -172,18 +182,50 @@ protected void setClosedLoop(boolean closedLoop) {
...
+     * @param yPower     the y power
+     * @param thetaPower the theta power
+     * @param moduleToTurnAround the module to turn around
      */
-    protected void fieldRelativeDrive(double xPower, double yPower, Rotation2d targetAngle) {
-        targetAngle = AllianceUtils.toMirroredAllianceRotation(targetAngle);
-        final ChassisSpeeds speeds = selfRelativeSpeedsFromFieldRelativePowers(xPower, yPower, 0);
-        speeds.omegaRadiansPerSecond = calculateProfiledAngleSpeedToTargetAngle(targetAngle);
+    protected void fieldRelativeDriveRotateAroundModule(double xPower, double yPower, double thetaPower, ModuleUtils.ModuleName moduleToTurnAround){
+        final ChassisSpeeds speeds = selfRelativeSpeedsFromFieldRelativePowers(xPower, yPower, thetaPower);
```
Comment (YoniKiriaty) [inline src/main/java/frc/robot/subsystems/swerve/Swerve.java:193]: why final

File: src/main/java/frc/robot/subsystems/swerve/ModuleUtils.java (hunk 1 above)
Comment (YHerm) [reply src/main/java/frc/robot/subsystems/swerve/ModuleUtils.java:26]: you can't

File: src/main/java/frc/robot/subsystems/swerve/Swerve.java [hunk 3]
```diff
@@ -200,6 +201,16 @@ protected void rotateToAngle(Rotation2d targetAngle) {
         selfRelativeDrive(fieldRelativeSpeedsToSelfRelativeSpeeds(targetFieldRelativeSpeeds));
     }
 
+    protected void rotateToAngleAroundWheel(Rotation2d targetAngle, ModuleUtils.ModuleName moduleName) {
+        final ChassisSpeeds targetFieldRelativeSpeeds = new ChassisSpeeds(0,
```
Comment (YoniKiriaty) [inline src/main/java/frc/robot/subsystems/swerve/Swerve.java:205]: bro space

**Review by YoniKiriaty — APPROVED** [review]

**Pull request closed without merging**